### PR TITLE
Updating dockerode from 2.2.10 to 3.2.1

### PR DESCRIPTION
**Pull request by visargsoneji**
Breaking change:
Promises interface was standardized, output property was removed.

### CODE REVIEW

File: broker/applications/operators/src/docker-operator/DockerService.js [hunk 1]
```diff
@@ -790,7 +790,8 @@ class DockerService extends BaseService {
       .logsAsync({
         stdout: 1,
         stderr: 1,
-        timestamps: 1
+        timestamps: 1,
+        follow: true
```
Comment (visargsoneji) [inline broker/applications/operators/src/docker-operator/DockerService.js:794]: To get the logs of the container, we need to pass the `follow` parameter as true which is `false` by default.

File: broker/core/utils/src/commonFunctions.js [hunk 2]
```diff
@@ -598,6 +598,10 @@ function demux(stream, options) {
     stream.on('readable', onreadable);
     stream.once('end', onend);
     stream.once('error', onerror);
+
+    setTimeout(function() {
+      stream.destroy();
+    }, 2000);
```
Comment (visargsoneji) [inline broker/core/utils/src/commonFunctions.js:604]: If we are not destroying the stream, it gets stuck causing `502` Bad Gateway error. So, destroying it was a necessary step.

Comment (jayeshjk) [reply broker/core/utils/src/commonFunctions.js:604]: @visargsoneji  can you pls add examples from dockerode which you referred for this as a comment. Might be helpful as future reference.

File: broker/applications/operators/src/docker-operator/DockerService.js (hunk 1 above)
Comment (jayeshjk) [reply broker/applications/operators/src/docker-operator/DockerService.js:794]: Can you pls add a reference here as well/

**Comment by visargsoneji** [comment]
Reference example from dockerode : https://github.com/apocas/dockerode/blob/master/examples/logs.js
`function containerLogs(container) {

  // create a single stream for stdin and stdout
  var logStream = new stream.PassThrough();
  logStream.on('data', function(chunk){
    console.log(chunk.toString('utf8'));
  });

  container.logs({
    follow: true,
    stdout: true,
    stderr: true
  }, function(err, stream){
    if(err) {
      return logger.error(err.message);
    }
    container.modem.demuxStream(stream, logStream, logStream);
    stream.on('end', function(){
      logStream.end('!stop!');
    });

    setTimeout(function() {
      stream.destroy();
    }, 2000);
  });
}`

**Review by jayeshjk — APPROVED** [review]
LGTM

**Review by subhankarc — APPROVED** [review]

**Pull request merged**